### ISSUE TITLE
About the following code

### ISSUE THREAD

**Issue by ziyingsk** [baoguangsheng/fast-detect-gpt#4]
In fast_detect_gpt.py

log_likelihood = lprobs.gather(dim=-1, index=labels) 
 (lprobs indexed from score vocabulary, while labels  indexed from reference vocabulary)

vocab_size = min(logits_ref.size(-1), logits_score.size(-1))
logits_ref = logits_ref[:, :, :vocab_size]
logits_score = logits_score[:, :, :vocab_size]

This indicates that the reference model should have almost the same vocabulary size (50,257 in your case) and index  as the score model. I think this should be mentioned in the paper, because this is a key condition to be so fast. Again, maybe the relative sentence is somewhere and I missed it. Actually, maybe this sentence should be in the abstract.

Sorry that I give strict comments to your paper. It is a good paper and I really like it!!

**Comment by baoguangsheng** [comment]
The key to the efficiency of Fast-DetectGPT is the independent sampling of alternative tokens. To score the sampled tokens, it is pre-requisite for the scoring model to have the same vocabulary as the sampling model. This condition is naturally satisfied when we employ the same model for sampling and scoring, as demonstrated in our experiments across 5-model generations.

However, it is possible to explore separate models for sampling and scoring, as our experiments with ChatGPT/GPT-4 generations. In such cases, it is essential to ensure that the two models share the same vocabulary. Equation 5 in the paper signifies this prerequisite condition, as both the model $q_{\phi}$ and $p_{\theta}$ utilize the same $\tilde{x}_i$. We intend to update the paper to explicitly address this aspect in the description.
